txcache/accountBreadcrumb.go

```diff
@@ -38,3 +39,70 @@ func (breadcrumb *accountBreadcrumb) updateNonce(lastNonce core.OptionalUint64)
 		breadcrumb.lastNonce = lastNonce
 	}
 }
+
+func (breadcrumb *accountBreadcrumb) createOrUpdateVirtualRecord(
+	virtualAccountsByAddress map[string]*virtualAccountRecord,
+	accountState state.UserAccountHandler,
+	address string,
+) {
+	virtualRecord, ok := virtualAccountsByAddress[address]
+	if !ok {
+		initialBalance := accountState.GetBalance()
+		virtualRecord = newVirtualAccountRecord(breadcrumb.initialNonce, initialBalance)
+		virtualAccountsByAddress[address] = virtualRecord
+	}
+
+	virtualRecord.updateVirtualRecord(breadcrumb)
+}
+
+func (breadcrumb *accountBreadcrumb) isContinuous(
+	address string,
+	accountNonce uint64,
+	sendersInContinuityWithSessionNonce map[string]struct{},
+	accountPreviousBreadcrumb map[string]*accountBreadcrumb,
+) bool {
+	if breadcrumb.isRelayer() {
+		return true
+	}
+
```

Comment:
Oh, so here you have distinct flows for continuity wrt. non-virtual-session (base) vs. continuity wrt. previous breadcrumb, sorry :)